src/main/java/io/anserini/index/IndexCollection.java

```diff
@@ -400,6 +449,121 @@ private void flush() {
 
   }
 
+  private final class ESIndexerThread implements Runnable {
+    private final Path input;
+    private final DocumentCollection collection;
+    private BulkRequest bulkRequest;
+
+    private ESIndexerThread(DocumentCollection collection, Path input) {
+      this.input = input;
+      this.collection = collection;
+      this.bulkRequest = new BulkRequest();
+    }
+
+    @Override
+    public void run() {
+      try {
+
+        LuceneDocumentGenerator generator = (LuceneDocumentGenerator) generatorClass.getDeclaredConstructor(Args.class, Counters.class).newInstance(args, counters);
+        BaseFileSegment<SourceDocument> iter = (BaseFileSegment) ((SegmentProvider) collection).createFileSegment(input);
+
+        int cnt = 0;
+        while (iter.hasNext()) {
+          SourceDocument sourceDocument;
+          try {
+            sourceDocument = iter.next();
+          } catch (RuntimeException e) {
+            counters.skipped.incrementAndGet();
+            continue;
+          }
+
+          if (!sourceDocument.indexable()) {
+            counters.unindexable.incrementAndGet();
+            continue;
+          }
+
+          // Used for indexing distinct shardCount of a collection
+          if (args.shardCount > 1) {
+            int hash = Hashing.sha1().hashString(sourceDocument.id(), Charsets.UTF_8).asInt() % args.shardCount;
+            if (hash != args.shardCurrent) {
+              counters.skipped.incrementAndGet();
+              continue;
+            }
+          }
+
+          Document document = generator.createDocument(sourceDocument);
+          if (document == null) {
+            counters.unindexed.incrementAndGet();
+            continue;
+          }
+          if (whitelistDocids != null && !whitelistDocids.contains(sourceDocument.id())) {
+            counters.skipped.incrementAndGet();
+            continue;
+          }
+
+          XContentBuilder builder = XContentFactory.jsonBuilder();
+          builder.startObject();
+          for (IndexableField field : document.getFields()) {
+            if (field.stringValue() != null) {
+              builder.field(field.name(), field.stringValue());
+            } else if (field.numericValue() != null) {
+              builder.field(field.name(), field.numericValue());
+            }
+          }
+          builder.endObject();
+          bulkRequest.add(new IndexRequest(input.getFileName().toString(), "doc", sourceDocument.id()).source(builder));
```

Comment:
When documents are indexed, they are placed in an index based on the name of the file they came from. Can we add a -es.index option (just like -solr.index) to specify the name of the index?

![image](https://user-images.githubusercontent.com/1282383/58733150-a44dce00-83c1-11e9-99c4-bd62ac863999.png)